src/vm_pages.rs

```diff
@@ -1481,14 +1481,18 @@ impl<'a, T: GuestStagePagingMode> FinalizedVmPages<'a, T> {
         let invalidated = self
             .inner
             .root
-            .invalidate_range(page_addr, num_pages * PageSize::Size4k as u64, |addr| {
-                self.inner.page_tracker.is_mapped_page(
-                    addr,
-                    PageSize::Size4k,
-                    self.inner.page_owner_id,
-                    P::mem_type(),
-                )
-            })
+            .invalidate_range(
+                page_addr,
+                num_pages * PageSize::Size4k as u64,
+                |addr, ps| {
+                    self.inner.page_tracker.is_mapped_page(
+                        addr,
+                        ps,
```

Comment:
we always convert pages at 4k granularity since the host is exclusively mapped with 4k pages.

Reply:
well, we should reject any `ps` that's not 4k

Reply:
this applies to everywhere else we're getting pages from the host.